python/lsst/ts/observatory/control/base_tcs.py

```diff
@@ -1301,6 +1471,100 @@ async def find_target(self, az, el, mag_limit, mag_range=2.0, radius=0.5):
 
         return f"{target_main_id}".rstrip()
 
+    async def find_target_local_catalog(
+        self, az, el, mag_limit, mag_range=2.0, radius=0.5
+    ):
+        """Make a cone search in the internal catalog and return a target
+        close to the specified position.
```

Comment:
Contrary to the `find_target_simbad` function, this function returns any target close to the specified position (as opposed to only targets from the HD catalog). I know that only the HD catalog PD file currently is present but this could change in the future. It worries me that this distinction between the two functions exists and I suggest that this is documented well here. Also, the magnitude limit and range are given but there is no check if the range falls outside of the catalog contents (which only has stars brighter than magnitude 6).

Reply:
So, `find_target_simbad` does not limit the search to the HD catalog, as I explained above. This one does a search in a "local catalog" loaded by the user. The catalog that ships with the module is only a test catalog, and I plan on injecting the full HD catalog and probably Gaia in the future. But I am probably not going to ship that with the module, but rather add them after the installation. 

The method does check If there are targets in the magnitude range and report an error in case there are none, see lines 1519-1521.

Reply:
Thanks for the explanation regarding the magnitude range check.